includes/abstracts/abstract-wc-legacy-product.php

```diff
@@ -295,8 +484,82 @@ protected function get_availability_class() {
 	 */
 	public function __isset( $key ) {
 		return metadata_exists( 'post', $this->id, '_' . $key );
+
+		// @todo handle variation meta here too
+		if ( in_array( $key, array_keys( $this->variation_level_meta_data ) ) ) {
+			return metadata_exists( 'post', $this->variation_id, '_' . $key );
+		} elseif ( in_array( $key, array_keys( $this->variation_inherited_meta_data ) ) ) {
+			return metadata_exists( 'post', $this->variation_id, '_' . $key ) || metadata_exists( 'post', $this->id, '_' . $key );
+		} else {
+			return metadata_exists( 'post', $this->id, '_' . $key );
+		}
 	}
 
+	/*
```

Comment:
Commented out code block. We should remove this unless its needed for something, so we don't end up committing a big block of commented out code.